src/main/java/org/testng/TestNG.java

```diff
@@ -1368,11 +1376,11 @@ private SuiteRunner createSuiteRunner(XmlSuite xmlSuite) {
         m_testRunnerFactory,
         m_useDefaultListeners,
         m_methodInterceptors,
-        m_invokedMethodListeners,
-        m_testListeners,
-        m_classListeners);
+        Lists.newArrayList(m_invokedMethodListeners.values()),
```

Comment:
We can change or add a new signature for `SuiteRunner` constructor with collections instead of lists. It will avoid useless objects creation.